sdk-lib/src/main/rust/lib.rs

```diff
@@ -138,29 +140,22 @@ pub unsafe extern "C" fn Java_cash_z_ecc_android_sdk_jni_RustBackend_initAccount
             .map(|i| env.get_object_array_element(ufvks_arr, i))
             .map(|jstr| utils::java_string_to_rust(&env, jstr.unwrap().into()))
             .map(|ufvkstr| {
-                // TODO: replace with `zcash_address::unified::Ufvk`
-                utils::fake_ufvk_decode(&ufvkstr).ok_or_else(|| {
+                UnifiedFullViewingKey::decode(&network, &ufvkstr).map_err(|e| {
+                    if e.starts_with("UFVK is for network") {
```

Comment:
This string matching is a bit ugly; can `UnifiedFullViewingKey::decode` return an enum for its error type instead?

Reply:
Yeah, we can rework this API once we've finished working out how it is used. I was more focused on other parts of the stack to get us to this point.

Reply:
Please file an issue so that we don't lose track.